src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -240,23 +262,15 @@ const submitEncryptModeForm = async (
         })
       }
       uinFin = jwtPayloadResult.value.userName
+      uinFin = form.isNricMaskEnabled ? maskNric(uinFin) : uinFin
+      parsedResponses.addNdiResponses({
+        authType,
+        uinFin,
+      })
       break
     }
   }
 
-  // Mask if Nric masking is enabled
```

Comment:
this may be a subjective thing, but one thing i felt was good about `encrypt-submission.controller` over `email-submission.controller` was that the controller logic was done step by step.

for eg, fetch `uinFin`, `userInfo` step -> mask nric step -> create email payload step
this makes the code easier to update and reduces duplicate function calls. 
the benefit in the future is that as the controller logic grows in size in the future, it is easier to move some of these steps into middleware functions (Eg, generate email payload middleware, singpass mask middleware) instead of increasing the complexity of `submitEncrpytForm` over time. 

what do you think about keeping the implementation to extractAuthinfo -> mask nric -> generatePayload? 
though i understand that this would cause duplicate switch statements (but i think its better to duplicate control flow rather than business logic)

Reply:
but otherwise, i think the current implementation also works. the above is just a discussion topic I felt is worth discussing

Reply:
Good point!